runtime/signals_nat.c

```diff
@@ -74,7 +74,11 @@ extern char caml_system__code_begin, caml_system__code_end;
 
 void caml_garbage_collection(void)
 {
-  caml_young_limit = caml_young_trigger;
+  /* TEMPORARY: if we have just sampled an allocation in native mode,
+     we simply renew the sample to ignore it. Otherwise, renewing now
+     will not have any efect on the sampling distribution, because of
+     the memorylessness of the Poisson process. */
+  caml_memprof_renew_minor_sample();
```

Comment:
`caml_memprof_renew_minor_sample` is not changed by `caml_memprof_renew_minor_sample();`, so I do not see what is the problem you are speaking about.

Are you speaking about `caml_memprof_young_trigger`? Then, yes, indeed, the value is lost. To see why this is correct, we have to distinguish two cases:
- Either the allocation that called `caml_garbage_collection` did not trigger the allocation. Then, yes, we will forget the position of the next sample, but, because of the memorylessness of the exponential distribution, the distribution of the next sample  will be the same.
- Either it triggered it. Temporarily, sampling in native code is not supported (in this PR). So, the only thing we can do is ignoring this sample. We do this by generating a new sample, and hope that it will not happen to re-sample the same block (otherwise we will just retry... no real harm here).

Reply:
I was thinking of the write to `caml_young_trigger` being not-written and a trigger going missing, but indeed now I can see (again) that `set_trigger` does the right thing. Sorry for the noise.